dagmc_stats/DagmcQuery.py

```diff
@@ -593,3 +593,48 @@ def __calc_tri_roughness(self):
             row_data = {'tri_eh': tri, 'roughness': rval}
             tri_roughness.append(row_data)
         self.__update_tri_data(tri_roughness)
+
+    def add_tag(self, tag_name, tag_type, tag_dic=None):
+        """Add tag according to given tag information
+
+        inputs
+        ------
+            tag_name : tag name
+            tag_type : tag type
+            tag_dic : a dictionary containing eh:data
+            
+        
+        outputs
+        -------
+            none
+        """
+        # the default tag size is 1, else is len(dic_vals)
+        tag_size = 1
+        
+        # check if lengths of values in the tag_dic are consistent
+        if tag_dic is not None:
+            # integer elements in tag_dic.values()
+            int_val = list(filter(lambda val: type(val) is not list, tag_dic.values()))
+            # list elements in tag_dic.values()
+            lst_val = [val for val in tag_dic.values() if val not in int_val]
+            if len(lst_val) != 0:
+                tag_size = len(lst_val[0])
+
+            if len(int_val) != 0 and len(lst_val) != 0 or \
+                    len(lst_val) != 0 and not all(len(lst_val) == tag_size for val in tag_dic.values()):
```

Comment:
This seems more complicated than we need.  If I understand correctly, `tag_dic.values()` has two valid options.  They are either all scalars, or all lists of the same size.

Can we just check those criteria, one at a time?
1. count the number of entries, E, that are lists (it should either be 0 or the same as the length of the values())
2. if 0 < E < N - error/warn
3. if E==0:  get the length of the first entry and check that the remainder are all that length

Reply:
This currently probably does the same thing, but it is not clear to read/follow.

Reply:
I think in this way the situation where the list entries are of different lengths (e.g. ([1,2,3], [1,2], [1,2]) are not considered. I would still prefer the current way.